x-pack/plugins/apm/public/hooks/use_table_pagination.ts

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  CriteriaWithPagination,
+  EuiTableSortingType,
+  Pagination,
+} from '@elastic/eui';
+import { orderBy } from 'lodash';
+import { useMemo, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import uuid from 'uuid';
+import { fromQuery, toQuery } from '../components/shared/links/url_helpers';
+import { useApmParams } from './use_apm_params';
+
+interface Initials<T extends any[]> {
+  // totalItemCount is not necessary here
+  initialPagination: Omit<Pagination, 'totalItemCount'>;
+  initialSort?: EuiTableSortingType<T[0]>;
+}
+
+type Props<T extends any[]> = {
+  items: T;
+  urlState?: boolean;
+} & Initials<T>;
+
+const PAGE_SIZE_OPTIONS = [10, 25, 50];
```

Comment:
I think we have a page size of 5 on the tables on the service overview, should we add this value here too?

Reply:
@MiriamAparicio the **ManagedTable** uses the `[10, 25, 50]` as default, I think we should keep with the same value and anything different than that should be a custom option. Later on, if we see that we should add `5` in the default value we could do it though.